### PR TITLE
Fix emerynet's faucet's flakyness

### PR DESCRIPTION
**Pull request by usmanmani1122**
## Description
Issue: [Fix emerynet's faucet's flakyness](https://github.com/orgs/Agoric/projects/19/views/1?filterQuery=assignee%3Ausmanmani1122&pane=issue&itemId=75006299)

This PR fixes the flakiness of faucet. The reason for the flakiness is usage of `block` broadcast mode for the transactions. This mode waits for the transaction to be included in a block before returning a response which can timeout in some cases. More details can be found in the linked issue.

The flow is now updated to use `sync` broadcast mode. The sync mode only waits for [CheckTx](https://docs.cosmos.network/main/learn/advanced/baseapp#checktx-state-updates) validation (without waitng for [DeliverTx](https://docs.cosmos.network/v0.47/learn/beginner/tx-lifecycle#delivertx)). The user is then navigated to a status page which polls for the transaction to be added in a block. For provisioning case, provisioning is handled asynchronously in the background (triggered on transaction success) as this doesn't effect the user
#### Extra changes:
- Bumped `eslint-plugin-jsdoc` package as the current version doesn't support node versions above `18`
- Bumped `express` due to a security vulnerbility [CVE-2024-29041](https://nvd.nist.gov/vuln/detail/CVE-2024-29041)

### CODE REVIEW

File: bases/shared/instagoric-server/server.js [hunk 1]
```diff
@@ -381,64 +388,150 @@ const addRequest = (address, request) => {
...
-              exitCode = await $`\
-            ${agBinary} tx swingset provision-one faucet_provision ${address} ${clientType} \
-  -b block --from ${FAUCET_KEYNAME} \
-  -y --keyring-backend test --keyring-dir=${agoricHome} \
-  --chain-id=${chainId} --node http://localhost:26657 \
-`.exitCode;
-            }
+            [exitCode, txHash] = await sendFunds(address, CLIENT_AMOUNT);
+            if (!exitCode)
+              pollForProvisioning(address, clientType, txHash);
```
Comment (frazarshad) [inline bases/shared/instagoric-server/server.js:511]: quick question: shouldnt this polling also happen in our transaction status endpoints?

Comment (usmanmani1122) [reply bases/shared/instagoric-server/server.js:511]: Didn't exactly understand your question.
Do you mean the `/transaction-status/:txhash` endpoints?

Comment (frazarshad) [reply bases/shared/instagoric-server/server.js:511]: yes.
because im assuming we're moving all types of polling to that endpoint to avoid timeouts

Comment (usmanmani1122) [reply bases/shared/instagoric-server/server.js:511]: So if I am getting it right, you are asking why we are not polling for the completion of the provisioning operation?
If not, can you elaborate your question?

**Review by frazarshad — APPROVED** [review]

**Pull request merged**